specs/agents/tracing-instrumentation-aws-lambda.md

```diff
@@ -0,0 +1,158 @@
+`context.message.body` |  | The message body. Sould only be captured if body capturing is enabled in the configuration.</br></br> Use `record.body` | The message body. Sould only be captured if body capturing is enabled in the configuration.</br></br> Use `record.sns.message`
+`context.message.headers` |  | The message attributes. Should only be captured, if capturing headers is enabled in the configuration.</br></br> Use `record.messageAttributes`| The message attributes. Should only be captured, if capturing headers is enabled in the configuration.</br></br> Use `record.sns.messageAttributes`
+
+
+### S3
+Lambda functions that are triggered by S3 accept an `event` input that may contain one ore more `S3 event notification records` in the `event.records` array. We cannot (automatically) wrap the processing of the individual records that are sent as a batch of S3 event notification records with a single `event`. 
+
+Thus, in case that an S3 `event` contains **exactly one** `S3 event notification record`, the agents must apply the following, S3-specific retrieval of information. Otherwise, the agents should apply the [Generic Lambda Instrumentation](generic-lambda-instrumentation) as desribed above.
+
+In addition the following fields should be set for Lambda functions triggered by S3:
+Field | Value | Description 
+---   | ---   | ---
+`faas.trigger.type` | `datasource` | Constant value for message based triggers.
+`faas.trigger.name` | e.g. `mybucket` | Use `record.s3.bucket.name`
+`faas.trigger.id` | e.g. `arn:aws:s3:::mybucket/ObjectCreated:Put` | Use `record.s3.bucket.arn`
+`faas.trigger.region` | e.g. `us-east-1` | Use `record.awsRegion`
+`faas.trigger.version` | e.g. `2.1` | Use `record.eventVersion`
+`faas.execution` | e.g. `arn:aws:s3:::mybucket/ObjectCreated:Put` | Format: `${record.s3.bucket.arn}/${record.eventName}`
+
+
+## Init spans
+With both API Gateway versions the `event` objects (may) contain a timestamp of the original API Gateway request. The difference between the API Gateway timestamp and the actual start timestamp of the Lambda function is a good estimate for the initialization phase of the lambda function. 
+The agents should use this information (if available) to adapt the transaction and spans structure of the Lambda invocation to represent the initialization time:
+
+```
+...
+  [XXXXXXXXXXXXX POST api.gateway XXXXXXXXXXXXXXXXXXXXX]   // client calling the API Gateway
+     [XXXXXXXXXX MyLambdaFunction XXXXXXXXXXXXXXXXXXXX]    // Lambda Transaction
+     [XXX Lambda init XXXXX]                               // Init span
```

Comment:
I'm unconvinced that we should be creating an init span. As @trentm pointed out there's a possibility that this could be misleading, as there could also be blocking wait time in addition to actual init time. I think just starting the transaction at the API Gateway timestamp will show this delay and not make assumptions over what *caused* that delay.

Thoughts?

Reply:
Good point, see comment / discussion above.